### PR TITLE
Rewrite ReplayStateMachine

### LINKED ISSUE #7430: Build ReplayStateMachine
**Description**

Based on https://github.com/zeebe-io/enhancements/pull/16 we need a replay state machine which replays events only. This should be created in parallel to the ReprocessingStateMachine first. Afterwards we can replace it completelty.

See related POC https://github.com/camunda-cloud/zeebe/commits/zell-7328-poc-state-on-followers and code https://github.com/camunda-cloud/zeebe/blob/zell-7328-poc-state-on-followers/engine/src/main/java/io/camunda/zeebe/engine/processing/streamprocessor/ReplayStateMachine.java

### LINKED ISSUE #7431: Replace ReprocessingStateMachine with ReplayStateMachine
**Description**

Based on zeebe-io/enhancements#16 we need a replay state machine which replays events only. 

Blocked by https://github.com/camunda-cloud/zeebe/issues/7430

See related POC https://github.com/camunda-cloud/zeebe/commits/zell-7328-poc-state-on-followers and code https://github.com/camunda-cloud/zeebe/blob/zell-7328-poc-state-on-followers/engine/src/main/java/io/camunda/zeebe/engine/processing/streamprocessor/ReplayStateMachine.java
A clear and concise description of what this issue is about.

### PR DESCRIPTION
**Pull request by ChrisKujawa**
## Description

Rewrites the ReplayStateMachine. I tried to simplified as far as I could. I removed the scan and the blacklist check. I fixed an issue with the reading of metadata and ignoring of an exception. 

## Related issues

closes #7430 
closes https://github.com/camunda-cloud/zeebe/issues/7431

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [ ] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/zeebe-io/zeebe/compare/stable/0.24...develop?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/0.25`) to the PR, in case that fails you need to create backports manually.

Testing:
* [ ] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark 

Documentation: 
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] New content is added to the [release announcement](https://drive.google.com/drive/u/0/folders/1DTIeswnEEq-NggJ25rm2BsDjcCQpDape)

### CODE REVIEW

File: engine/src/main/java/io/camunda/zeebe/engine/processing/streamprocessor/ReplayStateMachine.java [hunk 1]
```diff
@@ -28,55 +28,17 @@
...
- *                    |                          |       |                      |
- *                    +------^------------+------+       +---^------------+-----+
- *                           |            |                  |            |
- *                           +------------+                  +------------+
- *                             exception                       exception
- * </pre>
- *
- * See https://textik.com/#773271ce7ea2096a
- */
+/** Represents the state machine to replay events and rebuild the state. */
```
Comment (Zelldon) [inline engine/src/main/java/io/camunda/zeebe/engine/processing/streamprocessor/ReplayStateMachine.java:31]: Should be improved by https://github.com/camunda-cloud/zeebe/issues/7476

File: engine/src/main/java/io/camunda/zeebe/engine/processing/streamprocessor/ReplayStateMachine.java [hunk 2]
```diff
@@ -136,201 +98,166 @@ public ReplayStateMachine(final ProcessingContext context) {
...
+  }
+
+  /**
+   * Ends the replay and sets some important properties, especially completes the replay future with
+   * the last source event, which has caused the last applied event.
+   */
+  private void endReplay() {
+    // Replay ends at the end of the log
+    // reset the position to the first event where the processing should start
+    // TODO(zell): this should probably done outside; makes no sense here
```
Comment (Zelldon) [inline engine/src/main/java/io/camunda/zeebe/engine/processing/streamprocessor/ReplayStateMachine.java:160]: Should be done via https://github.com/camunda-cloud/zeebe/issues/7477

File: engine/src/main/java/io/camunda/zeebe/engine/processing/streamprocessor/ReplayStateMachine.java [hunk 3]
```diff
@@ -136,201 +98,166 @@ public ReplayStateMachine(final ProcessingContext context) {
...
-
-    recoveryFuture.complete(lastProcessedPosition);
+  /**
+   * Commits the current transaction. The transaction holds the state changes, which have been
+   * caused by applying the current event.
+   *
+   * @return true on success
+   * @throws Exception if the commit fails
+   */
+  private boolean commitCurrentAppliedStateChanges() throws Exception {
```
Comment (saig0) [inline engine/src/main/java/io/camunda/zeebe/engine/processing/streamprocessor/ReplayStateMachine.java:258]: (optional) We could rename the method to `tryToCommitStateChanges()`.

This name would align with `tryToApplyCurrentEvent()`.

**Review by saig0 — APPROVED** [review]
I like it :+1: It looks a way better than before :sparkles: 

I've one minor issue and some optional suggestions.  
Please have a look before merging :cake:

**Comment by Zelldon** [comment]
Thanks @saig0 for your review. Constructive and helpful comments. I applied all hints. Will merge it now.

bors r+

**Pull request merged**